draft-sharabayko-mops-srt.md

```diff
@@ -2086,15 +2086,584 @@ and RTTVar are updated by the socket based on algorithms for the sender
+two groups of congestion control algorithms defined in SRT: live
+congestion control (LiveCC) and file transfer congestion control
+(FileCC).
+
+## SRT Packet Pacing and Live Congestion Control (LiveCC) {#liveCC}
+
+To ensure smooth video playback of a receiving peer, SRT in live mode
+({{transmission-mode-live}}) must control the sender's buffer level to
+prevent overfill and depletion. Pacing control module is
+designed to send packets as fast as they are submitted by the
+application while maintaining relatively stable buffer level.
+While this looks like a simple problem, the sender's constituent
+of the Automatic Repeat Request (ARQ) system between
+input and output of the SRT sender adds some complexity.
+
+SRT needs a certain amount of bandwidth overhead in order to have
+space to insert retransmitted packets without affecting the SRT
+sender's main stream output rate too much so that packets can be
+transmitted normally.
+
+It is done by adjusting the maximum allowed bandwidth MAX_BW
+({{maxbw}}) which limits the bandwidth usage by SRT and is used
+by live congestion control (LiveCC) module to calculate the minimum
+inter-sending time of consecutive packets PKT_SND_PERIOD.
+In principle, the space between packets determines where
+retransmissions can be inserted, and the overhead represents the
+available margin. There is an empiric calculation that defines the
+interval, in microseconds, between two packets to give a certain
+bitrate. It's a function of the average packet payload (which
+includes video, audio, etc.) and configured maximum bandwidth MAX_BW.
+See {{default-liveCC}} for details.
+
+In live mode, it is allowed to drop those packets that can not be delivered
+in time ({{too-late-packet-drop}}).
+
+To sum up, a combination of pacing control and live
+congestion control (LiveCC) modules controls SRT packets pacing based
+on the input rate and an overhead for packets retransmission in order
+to avoid congestion during fluctuations of the source bitrate.
+
+In addition to these mechanisms, in highly variable networks live mode
+fairness can be accomplished outside SRT by controlling the bitrate
+of the encoder at the input of the SRT sender. SRT provides a variety
+of network related statistics such as RTT estimation, the level of
+packet loss, the number of packets dropped by SRT, etc., which can be
+used for making decisions and adjusting the bitrate in real time.
+
+### Configuring Maximum Bandwidth {#maxbw}
+
+There are several ways of configuring maximum bandwidth MAX_BW:
+
+1. By means of setting the value of maximum bandwidth MAX_BW
+   explicitly (MAXBW mode).
+
+   The recommended default value is 1 Gbps. Note that the default value
+   is set only for live transmission mode.
+
+   Note that this static setting is not well-suited to a variable input,
+   like when you change the bitrate on an encoder. Each time the input
+   bitrate is configured on the encoder, MAX_BW must also be reconfigured.
```

Comment:
We currently have for MAXBW mode the following explanation:

_The recommended default value is 1 Gbps. The default value
   is set only for live streaming.

   Note that this static setting is not well-suited to a variable input,
   like when you change the bitrate on an encoder. Each time the input
   bitrate is configured on the encoder, MAX_BW must also be reconfigured.

   In MAXBW mode, SRT does not follow any fluctuations in the encoder input
   making it unsuitable for variable bitrate (VBR) encoding._

If we change the second paragraph to your suggestion, the last one seems to be incorrect too. The same idea as you described. Do you have a suggestion how to change the last paragraph or maybe we should simply remove it, because it's already written in the second paragraph?

Reply:
Simply removing the paragraph about "In MAXBW mode, SRT does not follow any fluctuations.." is a good idea.